tests/test_string_accessors.py

```diff
@@ -71,40 +71,42 @@ def test_string_accessors_boolean_series():
 
 def test_string_accessors_boolean_index():
     idx = pd.Index(DATA)
-    _check = functools.partial(check, klass=np.ndarray, dtype=np.bool_)
-    _check(assert_type(idx.str.startswith("a"), np_ndarray_bool))
+    _check = functools.partial(check, klass=np_1darray[np.bool])
```

Comment:
```suggestion
    _check = functools.partial(check, klass=np_1darray[np.bool], dtype=np.bool_)
```
Then we will check the value in the array at runtime.